### PR TITLE
fix(conventional-changelog-conventionalcommits): optional config

### PR DESCRIPTION
**Pull request by lvjiaxuan**
It throws error, when the preset option is set to the string `conventionalcommits`.

image

### CODE REVIEW

File: packages/conventional-changelog-conventionalcommits/conventionalRecommendedBump.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ function createConventionalRecommendedBumpOpts (config, parserOpts) {
         }
       })
 
-      if (config.preMajor && level < 2) {
+      if (config?.preMajor && level < 2) {
```
Comment (dangreen) [inline packages/conventional-changelog-conventionalcommits/conventionalRecommendedBump.js:30]: Please add test case for that.

File: packages/conventional-recommended-bump/test/index.spec.js [hunk 2]
```diff
@@ -77,9 +77,7 @@ describe('conventional-recommended-bump', () => {
 
       const recommendation = await conventionalRecommendedBump({
         cwd: testTools.cwd,
-        preset: {
-          name: 'conventionalcommits'
-        }
+        preset: 'conventionalcommits'
       }, {})
```
Comment (dangreen) [inline packages/conventional-recommended-bump/test/index.spec.js:81]: Why did you change it?

**Review by dangreen — CHANGES_REQUESTED** [review]
Hi. Thanks for the PR!

File: packages/conventional-recommended-bump/test/index.spec.js (hunk 2 above)
Comment (lvjiaxuan) [reply packages/conventional-recommended-bump/test/index.spec.js:81]: Because this string preset causes the issue and performs the same thing as `{  name: 'conventionalcommits'  }`. If you think this is inappropriate, I will write an additional test case for it.

File: packages/conventional-recommended-bump/test/index.spec.js [hunk 3]
```diff
@@ -71,6 +71,15 @@ describe('conventional-recommended-bump', () => {
...
 
+  it('should allow the string `conventionalcommits` as the preset option', async () => {
+    preparing(2)
+
+    await expect(() => conventionalRecommendedBump({
+      cwd: testTools.cwd,
+      preset: 'conventionalcommits'
+    }, {})).not.toThrowError()
+  })
+
```
Comment (lvjiaxuan) [inline packages/conventional-recommended-bump/test/index.spec.js:82]: Finally, I decided to add a test case. 😁

**Review by dangreen — APPROVED** [review]
